pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -144,12 +144,24 @@ func (p *MellanoxPlugin) OnNodeStateChange(new *sriovnetworkv1.SriovNetworkNodeS
 		pciAddress := pciPrefix + "0"
 
 		// Skip devices not configured by the operator
-		if p.nicNotConfiguredByOperator(portsMap) {
+		isConfigured, err := p.nicConfiguredByOperator(portsMap)
+		if err != nil {
+			return false, false, err
```

Comment:
thats going to be a non-recoverable error right ? (also below at L160)

i guess if we get here then state file got corrupted. which "should not" happen

Reply:
what about first time running on the interface ? or after nodeState got re-created (we clearing the state file there) ?

this logic runs before generic plugin which eventually stores the pf status when it configures sriov interface

Reply:
That is right. 
Maybe add an error message and request the user to remove the file?

Reply:
+1

Reply:
done I put the info in the function itself (for both of them)